lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py

```diff
@@ -0,0 +1,184 @@
+from abc import abstractmethod
+from PIL import Image
+from typing import Any, Union, List, Dict
+from sycamore.data import BoundingBox, Element
+from sycamore.utils.cache import DiskCache
+from pathlib import Path
+from io import IOBase, BytesIO
+from sycamore.utils.pdf import pdf_to_image_files
+from sycamore.utils.import_utils import requires_modules
+from sycamore.transforms.text_extraction.text_extractor import TextExtractor
+import logging
+from sycamore.utils.time_trace import LogTime, timetrace
+import tempfile
+import os
+
+ocr_cache = DiskCache(str(Path.home() / ".sycamore/OCRCache"))
+
+logger = logging.getLogger(__name__)
+
+
+class OCRModel(TextExtractor):
+
+    @abstractmethod
+    def get_text(self, image: Image.Image) -> str:
+        pass
+
+    @abstractmethod
+    def get_boxes_and_text(self, image: Image.Image) -> List[Dict[str, Any]]:
+        pass
+
+    @timetrace("OCR Extraction")
+    def extract(self, filename: Union[str, IOBase], hash_key: str, use_cache=False, **kwargs) -> List[List[Element]]:
+        cached_result = ocr_cache.get(hash_key) if use_cache else None
+        if cached_result:
+            logger.info(f"Cache Hit for OCR. Cache hit-rate is {ocr_cache.get_hit_rate()}")
+            return cached_result
+        else:
+            with tempfile.TemporaryDirectory() as tempdirname:  # type: ignore
+                if isinstance(filename, IOBase):
```

Comment:
Why write out to a file here? Is this to communicate between the processes? I think all of the ocr models can take a file-like object.

Reply:
Changed it